scripts/deploy/github/kfp-readiness/wait_for_pods.py

```diff
@@ -21,15 +21,25 @@ def get_pod_statuses():
         pod_name = pod.metadata.name
         pod_status = pod.status.phase
         container_statuses = pod.status.container_statuses or []
-        ready_containers = sum(1 for status in container_statuses if status.ready)
-        total_containers = len(container_statuses)
-        statuses[pod_name] = (pod_status, ready_containers, total_containers)
+        ready = 0
+        total = 0
+        waiting_messages = []
+        for status in container_statuses:
+            total += 1
+            if status.ready:
+                ready += 1
+            if status.state.waiting is not None:
+                if status.state.waiting.message is not None:
+                    waiting_messages.append(f'CONTAINER: {status.name} - {status.state.waiting.reason}: {status.state.waiting.message}')
```

Comment:
alternatively to https://github.com/kubeflow/pipelines/pull/11284/files#r1795992913, perhaps we can add additional context to `CONTAINER` prefix such as `Waiting on Container {name} - {reason}: {message}`

Reply:
Sure will take this option.